proxy_tester_menu.py

```diff
@@ -1,41 +1,59 @@
 import contextlib
 import time
-import requests
 import os
+import argparse
+import concurrent.futures
+import subprocess
 
 DEFAULT_DOMAIN = "www.google.com"
 
 
 def test_proxy(ip, domain=DEFAULT_DOMAIN):
-    proxies = {
-        "http": f"http://{ip}",
-        "https": f"http://{ip}",
-    }
-    with contextlib.suppress(requests.RequestException):
-        response = requests.get(f"http://{domain}", proxies=proxies, timeout=5)
-        if response.status_code >= 200 and response.status_code < 300:
-            return True
-    return False
-
-
-def test_ip_addresses(ip_file, output_file, domain=DEFAULT_DOMAIN):
+    url = f"http://{domain}"
+    proxy_url = f"http://{ip}"
+    proxies = {'http': proxy_url, 'https': proxy_url}
+
+    try:
+        # Use subprocess to make requests using cURL
+        command = f"curl --connect-timeout 5 --max-time 5 --proxy {proxy_url} {url}"
```

Comment:
Using curl command line limits it only to system that have curl installed, mostly Linux systems.

Can you utilize urllib or http.client which comes with python in built library?